File: app/src/main/java/ie/dublinbuspal/android/data/remote/soap/xml/route/RouteXml.java
package ie.dublinbuspal.android.data.remote.soap.xml.route;

import org.simpleframework.xml.Element;
import org.simpleframework.xml.Root;

@Root(name = "Route", strict = false)
public class RouteXml {

    @Element(name = "Number", required = false)
    private String routeID;

    @Element(name = "From", required = false)
    private String routeStart;

    @Element(name = "Towards", required = false)
    private String routeEnd;

    public RouteXml() {
        super();
    }

    public String getRouteID() {
        return routeID;
    }

    public String getRouteStart() {
        return routeStart;
    }

    public String getRouteEnd() {
        return routeEnd;
    }

    public void setRouteID(String routeID) {
        this.routeID = routeID;
    }

    public void setRouteStart(String routeStart) {
        this.routeStart = routeStart;
    }

    public void setRouteEnd(String routeEnd) {
        this.routeEnd = routeEnd;
    }

}

File: service/src/main/kotlin/ie/dublinbuspal/service/model/route/RouteXml.kt
package ie.dublinbuspal.service.model.route

import org.simpleframework.xml.Element
import org.simpleframework.xml.Root

@Root(name = "Route", strict = false)
data class RouteXml(
        @field:Element(name = "Number") var id: String? = null,
        @field:Element(name = "From", required = false) var origin: String? = null,
        @field:Element(name = "Towards", required = false) var destination: String? = null
)
